include/dba.php

```diff
@@ -1415,38 +1303,44 @@ function dbesc($str) {
 //                   'user', 1);
 function q($sql) {
 	global $db;
+
 	$args = func_get_args();
 	unset($args[0]);
 
-	if ($db && $db->connected) {
-		$sql = $db->clean_query($sql);
-		$sql = $db->any_value_fallback($sql);
-		$stmt = @vsprintf($sql,$args); // Disabled warnings
-		//logger("dba: q: $stmt", LOGGER_ALL);
-		if ($stmt === false)
-			logger('dba: vsprintf error: ' . print_r(debug_backtrace(),true), LOGGER_DEBUG);
+	if (!$db || !$db->connected) {
+		return false;
+	}
 
-		$db->log_index($stmt);
+	$sql = $db->clean_query($sql);
+	$sql = $db->any_value_fallback($sql);
 
-		return $db->q($stmt);
+	$stmt = @vsprintf($sql, $args);
+
+	$ret = dba::p($stmt);
+
+	if (is_bool($ret)) {
+		return $ret;
 	}
 
-	/**
-	 *
-	 * This will happen occasionally trying to store the
-	 * session data after abnormal program termination
-	 *
-	 */
-	logger('dba: no database: ' . print_r($args,true));
-	return false;
+	$columns = dba::columnCount($ret);
+
+	$data = dba::inArray($ret);
+
+	if ((count($data) == 0) && ($columns == 0)) {
+		return true;
+	}
+
+	return $data;
 }
 
 /**
- * @brief Performs a query with "dirty reads"
+ * @brief Performs a query with "dirty reads" - deprecated
```

Comment:
Which function should be used instead? Can you add it to the docs?

Reply:
```q``` **and** ```qu``` are both deprecated and should be replaced with the ```dba::``` functions like ```dba::select``` or ```dba::exists```.

Reply:
I think it would be good to have this information in the doxygen doc as well. So when someone stumbles over the "deprecated" they already have the answer to the question "what else should be used"

Reply:
I added some documentation.